### DISCUSSION TITLE
Mission Length

### DISCUSSION THREAD

**Discussion by aka-scorp** [Khopa/dcs_liberation#2204] [Q&A] [answered]
Hi,  we are running Liberation with the dedicated server and are having problems getting the mission to end automatically.  We have the mission length set to 120 minutes but after 2 1/4 hours it is still chugging away.  We have to manually stop and report the mission.

Both Liberation and the DS are on the same computer.

I have looked in the mission and can't see anything that might end the mission but there is a lot in there.

Any ideas?

Thanks
[AKA_Campaign.zip](https://github.com/dcs-liberation/dcs_liberation/files/8624965/AKA_Campaign.zip)
,

**Comment by Starfire13** [comment, marked as answer]
The mission length isn't intended to stop the server. It's for the autoplanner. You have to stop it yourself (or switch to a blank mission).

**Reply by aka-scorp** [reply]
Thanks Starfire, wasn't sure if we were doing something wrong.  Any chance that is something that could be added?  If not we will just keep ending the mission manually.  For single player it's not a problem but for multiplayer on a dedicated server it would be nice.

**Reply by Starfire13** [reply]
Wouldn't you guys just end the mission once everyone has landed? If it were automated, you'd either have the mission end too early, or have to leave the mission running and wait.

**Comment by aka-scorp** [comment]
The DS runs on a server that only a few members have access to so if the admin leaves early there is no one around to end the mission.  This is how we have been running our BoX campaign for 3 years and it works well for us.  There are time left message each hour and 15, 10, 5 and 1 minutes before the end of the mission.  As I said if the campaign is being run as single player or hosted in game for multiplayer it's not a problem but we like to play the thing out so that AI units get to do what they are supposed to.

We can work around this by adding our own timer and Liberation is a great tool so thanks for all the work that has gone into it.
